src/Servers/Kestrel/Transport.Libuv/src/Internal/LibuvConnection.cs

```diff
@@ -246,6 +239,26 @@ private void OnRead(UvStreamHandle handle, int status)
             }
         }
 
+        private void FireConnectionClosed()
+        {
+            // Guard against scheduling this multiple times
+            if (_connectionClosed)
```

Comment:
I see you're taking advantage of the single-threaded nature of libuv's event loop 👍